src/components/Markings/DrawingClassifier.js

```diff
@@ -0,0 +1,257 @@
+
+const mapStateToProps = (state, ownProps) => {
+
+    const subjectList = state.classifier.subjectLists[ownProps.workflow.id] || []
+    const subjectsSeenThisSession = state.classifier.seenThisSession[ownProps.workflow.id] || []
+
+    const seenSubjectIds = subjectsSeenThisSession.map(subject => subject.id)
+    const usableSubjects = subjectList.filter(subject => !seenSubjectIds.includes(subject.id))
+    return {
+        isSuccess: state.classifier.isSuccess,
+        isFailure: state.classifier.isFailure,
+        isFetching: state.classifier.isFetching,
+        guide: state.classifier.guide[ownProps.workflow.id] || {},
+        tutorial: state.classifier.tutorial[ownProps.workflow.id] || {},
+        needsTutorial: state.classifier.needsTutorial[ownProps.workflow.id] || false,
+        usableSubjects
+    }
+}
+
+const mapDispatchToProps = (dispatch) => ({
+    imageActions: bindActionCreators(imageActions, dispatch),
+    classifierActions: bindActionCreators(classifierActions, dispatch),
+    navBarActions: bindActionCreators(navBarActions, dispatch)
+})
+
+const PAGE_KEY = 'DrawingScreen'
+
+class DrawingClassifier extends Component {
+
+    constructor(props) {
+        super(props)
+
+        this.state = {
+            imageIsLoaded: false,
+            localImagePath: '',
+            isQuestionVisible: !props.needsTutorial
+        }
+
+        this.finishTutorial = this.finishTutorial.bind(this)
+        this.setQuestionVisibility = this.setQuestionVisibility.bind(this)
+    }
+
+    static renderNavigationBar() {
+        return <NavBar pageKey={PAGE_KEY} showBack={true} />;
+    }
+
+    componentDidMount() {
+        const { display_name, inPreviewMode, navBarActions, classifierActions } = this.props
+        navBarActions.setTitleForPage(display_name, PAGE_KEY)
+        classifierActions.setClassifierTestMode(inPreviewMode)
+        if (inPreviewMode) {
+          navBarActions.setNavbarColorForPage(Theme.$testRed, PAGE_KEY)
+        } else {
+          navBarActions.setNavbarColorForPageToDefault(PAGE_KEY)
+        }
+      }
+
+    componentDidUpdate(prevProps) {
+        if (!R.equals(prevProps.usableSubjects, this.props.usableSubjects) && this.props.usableSubjects[0]) {
+            this.props.imageActions.loadImageToCache(this.props.usableSubjects[0].display.src).then(localImagePath => {
+                this.setState({
+                    imageIsLoaded: true,
+                    localImagePath
+                })
+            })
+        }
+    }
+
+    setQuestionVisibility(isVisible) {
+        this.setState({ isQuestionVisible: isVisible })
+    }
+
+    finishTutorial() {
+        if (this.props.needsTutorial) {
+            this.props.classifierActions.setTutorialCompleted(this.props.workflow.id, this.props.project.id)
+        } else {
+            this.setQuestionVisibility(true)
+        }
+    }
+
+    render() {
+        if (this.props.isFetching || !this.props.isSuccess) {
```

Comment:
Is there still a way to display an error in the case `isFetching` is completed but `isSuccess` remains false, leaving the overlay spinning?